calico_node/tests/st/utils/workload.py

```diff
@@ -51,20 +54,72 @@ def __init__(self, host, name, image="busybox", network="bridge", ip=None, label
         """
         self.host = host
         self.name = name
+        self.network = network
 
         lbl_args = ""
         for label in labels:
             lbl_args += " --label %s" % (label)
 
         ip_option = ("--ip %s" % ip) if ip else ""
-        command = "docker run -tid --name %s --net %s %s %s %s" % \
-                  (name, network, lbl_args, ip_option, image)
+        command = "docker run -tid --name %s --net=none %s %s" % \
+                  (name, lbl_args, image)
+        # Change to --net none here, and use Calico CNI instead.
         docker_run_wl = partial(host.execute, command)
         retry_until_success(docker_run_wl)
-        self.ip = host.execute(
-            "docker inspect --format "
-            "'{{.NetworkSettings.Networks.%s.IPAddress}}' %s"
-            % (network, name))
+        #self.ip = host.execute(
+        #    "docker inspect --format "
+        #    "'{{.NetworkSettings.Networks.%s.IPAddress}}' %s"
+        #    % (network, name))
+        self.run_cni("ADD", ip=ip)
+
+    def run_cni(self, add_or_del, ip=None):
+        adding = (add_or_del == "ADD")
+        workload_pid = self.host.execute("docker inspect --format '{{.State.Pid}}' %s" % self.name)
+        container_id = self.host.execute("docker inspect --format '{{.Id}}' %s" % self.name)
+        ip_json = (",\"args\":{\"ip\":\"%s\"}" % ip) if (ip and adding) else ""
+        ip_args = ("CNI_ARGS=IP=%s " % ip) if (ip and adding) else ""
+        etcd_args = "ETCD_ENDPOINTS=http://%s:2379 " % get_ip()
+        if ETCD_SCHEME == "https":
+            etcd_args = ("ETCD_ENDPOINTS=https://%s:2379 " % ETCD_HOSTNAME_SSL +
```

Comment:
Probably fine to pass in environment variables, but that's not how it's typically done. Usually these will get set in the CNI network config json passed in over stdin.

Reply:
Are you sure? In the documentation and examples that I've seen, these env vars are still needed (except ARGS).

Reply:
E.g. see https://github.com/projectcalico/cni-plugin/blob/master/test_utils/utils.go#L227

Reply:
Sorry, I wasn't very clear :)

I was just referring to the etcd variables. Ones you linked are part of the core CNI spec and definitely passed in through the env.